pkg/controller/master/backup/backup.go

```diff
@@ -93,44 +110,72 @@ func (h *Handler) OnBackupChange(key string, vmBackup *harvesterv1.VirtualMachin
 		return nil, nil
 	}
 
+	target, err := settings.DecodeBackupTarget(settings.BackupTargetSet.Get())
+	if err != nil {
+		return nil, err
+	}
+
 	if isBackupReady(vmBackup) {
-		return nil, nil
+		// generate vm backup metadata and upload to backup target
+		return nil, h.uploadVMBackupMetadata(vmBackup, target)
 	}
```

Comment:
We may still do upload multiple times. For example when Harvester server restarts and syncs all the vmBackups. All the ready backups will get uploaded again. Can we have a condition to ensure it is uploaded once?

Reply:
How about we check whether remote data exists? If it exists and the data is same, we skip upload here.

Reply:
That may work too. I'm sure how much effort is needed for the check

Reply:
Updated it. Thank you.